### PR TITLE
Improve docker-compose tests stability

### PR DESCRIPTION
**Pull request by potiuk**
The docker-compose tests are failing occassionally and part of it is that we were manually waiting and querying for the compose to be ready. Since we moved to docker compose v2 only we can now remove the conditional docker-compose v1 code and use `--wait` and `--wait-timetout` built-in functionality and increase the timeout a bit. This should be helpful to avoid a race condition where the docker components are started but not healthy yet, the flag `--wait` waits not only for the processes to start but also for the health checks.

Also increased the timeout of waiting for dags to be in terminal state.

This should help with stability of the test.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by hussein-awala — APPROVED** [review]
Looks good

**Comment by potiuk** [comment]
Re-run it with full tests to make sure the tests are run actually :)

**Review by amoghrajesh — APPROVED** [review]
Ah nice fine. LGTM +1

File: docker_tests/test_docker_compose_quick_start.py [hunk 1]
```diff
@@ -134,61 +134,39 @@ def test_trigger_dag_and_wait_for_result(tmp_path_factory, monkeypatch):
...
-        if not success:
-            print("ERROR: Neither `docker compose` nor `docker-compose` is available")
-            sys.exit(1)
+        print("ERROR: `docker compose` not available. Make sure compose plugin is installed")
+        sys.exit(1)
     compose_command.extend(["--project-name", "quick-start"])
     run_command([*compose_command, "config"])
     run_command([*compose_command, "down", "--volumes", "--remove-orphans"])
+    wait_for_containers_timeout = os.getenv("WAIT_FOR_CONTAINERS_TIMEOUT", "500")
+    run_command([*compose_command, "up", "-d", "--wait", "--wait-timeout", wait_for_containers_timeout])
```
Comment (Taragolis) [inline docker_tests/test_docker_compose_quick_start.py]: `--wait-timeout` was added in [Compose v 2.17.0](https://github.com/docker/compose/releases/tag/v2.17.0) but I guess we use 2.14.x in our CI. In additional it also breaks `--timeout` flag until [Compose v 2.17.3](https://github.com/docker/compose/releases/tag/v2.17.3)

Comment (Taragolis) [reply docker_tests/test_docker_compose_quick_start.py]:
```console
Good version of Docker: 23.0.0.
Good version of docker-compose: 2.14.1
```

**Comment by Taragolis** [comment]
In general looks good except `--wait-timeout` which we do not support in the CI yet

In additional I tried to find possibility to replace subprocess + Docker by [python-on-whales](https://github.com/gabrieldemarmiesse/python-on-whales) which I personally found more resilient way to build applications/utils on top of docker stack. So I have a plan to replace subprocess in docker-compose tests and after have a look is it support every feature, so we could try to replace stuff in breeze

**Comment by potiuk** [comment]
> In general looks good except `--wait-timeout` which we do not support in the CI yet
> 
Ah yeah. That's why we keep the "min" version of docker in CI to make sure our "min requirements" are good. Yeah. I will remove the timeout altogether.

> In additional I tried to find possibility to replace subprocess + Docker by [python-on-whales](https://github.com/gabrieldemarmiesse/python-on-whales) which I personally found more resilient way to build applications/utils on top of docker stack. So I have a plan to replace subprocess in docker-compose tests and after have a look is it support every feature, so we could try to replace stuff in breeze

This is a cool project :). Thanks for finding it and pointing to it. It seems to very nicely address the need we have - where we do not want to use the API of docker, but docker CLI (and it seem to have support for podman and nerdctl out-of-the-box which is cool.

One small watch-out. We currently have the `run_command` implemented in  Breeze to support `--dry-run` and `--verbose` commands: both of them super useful for testing and understanding what's going on under the hood. Generally both print generall copy-pasteable command that Breeze runs so that you can run it manually and play with it. I hope we can make `python-on-whales` support similar feature (and if not - maybe we can contribute it before we start using it).

**Comment by Taragolis** [comment]
> One small watch-out. We currently have the run_command implemented in Breeze to support --dry-run and --verbose commands: both of them super useful for testing and understanding what's going on under the hood. Generally both print copy-pasteable command that Breeze runs so that you can run it manually and play with it (including env vars). I hope we can make python-on-whales support similar feature (and if not - maybe we can contribute it before we start using it).

Yep, I think better to inspect breeze (maybe I can do on the next week) and collect all stuff which we lost in case of switch to `python-on-whales` and after decide is it worthwhile, new features are cool unless you lost something important.

I have experience with replace something simple and legacy, like replace couple of files which manually generate command for subprocess or shell scripts which run commands + jq for parse output in this case it is cool. But with breeze it just might not as we expects and just make maintainers sad. E.g. in theory `--dry-run` could be replaced by simple fake cli command which just print command and provide it in [client_call](https://gabrieldemarmiesse.github.io/python-on-whales/docker_client/) but in practice we never know until try it

**Pull request merged**